apps/web/app/queries/protocols/use-protocol-balance.ts

```diff
@@ -0,0 +1,41 @@
+import { useMemo } from 'react';
+
+import { useQuery } from '@tanstack/react-query';
+import { ProtocolSlug } from '~/features/stacking/start-liquid-stacking/utils/types-preset-protocols';
+import { getNetworkInstanceByName } from '~/features/stacking/start-pooled-stacking/utils/utils-stacking-pools';
+import { createEmptyQueryOptions } from '~/queries/create-empty-query-options';
+import { createGetDAOBalanceQueryOptions } from '~/queries/protocols/dao/balance';
+import { createGetLisaBalanceQueryOptions } from '~/queries/protocols/lisa/balance';
+import { CreateProtocolBalanceQueryOptionsParams } from '~/queries/protocols/protocol-types';
+import { useStacksClient } from '~/queries/stacks/stacks-client';
+import { useLeatherConnect } from '~/store/addresses';
+import { useStacksNetwork } from '~/store/stacks-network';
+
+export function useProtocolBalance(slug: ProtocolSlug) {
+  const client = useStacksClient();
+  const { stacksAccount } = useLeatherConnect();
+  const address = stacksAccount?.address;
+
+  const network = useStacksNetwork();
+  const networkMode = getNetworkInstanceByName(network.networkName);
+
+  const queryOptions = useMemo(() => {
+    const options: CreateProtocolBalanceQueryOptionsParams = {
+      address,
+      client,
+      networkMode,
+      networkUrl: network.networkPreference.chain.stacks.url,
+    };
+
+    switch (slug) {
+      case 'stacking-dao':
+        return createGetDAOBalanceQueryOptions(options);
+      case 'lisa':
+        return createGetLisaBalanceQueryOptions(options);
+      default:
+        return createEmptyQueryOptions();
```

Comment:
haven't seen this technique before lol, if we need conditional queries, isn't it better to populate an array of useQueries? Which can be empty, rather than mocked?